src/video.cpp

```diff
@@ -254,8 +254,8 @@ int Video::open(String a_path, bool a_load_audio) {
 		if (av_hwframe_transfer_data(av_hw_frame, av_frame, 0) < 0)
 			_printerr_debug("Error transferring the frame to system memory!");
 
-		y_data.resize(av_hw_frame->linesize[0] * resolution.y);
-		u_data.resize((av_hw_frame->linesize[1] / 2) * (resolution.y / 2) * 2);
+		y_data = Image::create_empty(av_frame->linesize[0] , resolution.y, false, Image::FORMAT_R8);
+		u_data = Image::create_empty(av_frame->linesize[1]/2 , resolution.y/2, false, Image::FORMAT_RG8);
 		padding = av_hw_frame->linesize[0] - resolution.x;
 		av_frame_unref(av_hw_frame);
 	} 
```

Comment:
Here kind of the same, we transfer the hardware frame data from av_frame to av_hw_frame (I know, the naming should kind of be the opposite ^^"). If it works with av_frame, we can also skip the av_hwframe_transfer_data and use av_frame for deciding the padding.